shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -59,29 +60,45 @@ static Point getImageSizeFromStream(InputStream is) {
   }
 
   static RobolectricBufferedImage getImageFromStream(InputStream is) {
+    return getImageFromStream(null, is);
+  }
+
+  static RobolectricBufferedImage getImageFromStream(String fileName, InputStream is) {
     if (!initialized) {
       // Stops ImageIO from creating temp files when reading images
       // from input stream.
       ImageIO.setUseCache(false);
       initialized = true;
     }
 
+    String format = null;
     try {
       ImageInputStream imageStream = createImageInputStream(is);
       Iterator<ImageReader> readers = ImageIO.getImageReaders(imageStream);
-      if (!readers.hasNext()) return null;
+      if (!readers.hasNext()) {
+        return null;
+      }
 
       ImageReader reader = readers.next();
       try {
         reader.setInput(imageStream);
-        String format = reader.getFormatName();
+        format = reader.getFormatName();
         int minIndex = reader.getMinIndex();
         BufferedImage image = reader.read(minIndex);
         return RobolectricBufferedImage.create(image, ("image/" + format).toLowerCase());
       } finally {
         reader.dispose();
       }
     } catch (IOException e) {
+      if (FORMAT_NAME_PNG.equalsIgnoreCase(format) && e instanceof IIOException) {
+        String pngFileName = "(" + (fileName == null ? "not given PNG file name" : fileName) + ")";
+        System.err.println(
+            "Looks like your PNG file"
+                + pngFileName
+                + " has format problem that OpenJDK can't process correctly. You can check"
+                + " https://github.com/robolectric/robolectric/issues/6812 for potential"
+                + " solution.");
```

Comment:
Can the error message be closer to the format:

"The PNG file " + pngFileName " cannot be decoded. This may be due to an OpenJDK issue with certain PNG files. See https://github.com/robolectric/robolectric/issues/6812 for more details." 

Also, will this error message be printed if we try to encoding an empty stream, or a random stream of garbage? It would be nice to different between an otherwide valid PNG that is not supported by OpenJDK vs a completely invalid PNG altogether.

Reply:
For instance, can this check for the string "Invalid chunk length" in the IOException?

Reply:
I think we can check "Invalid chunk length". I will update it.

Reply:
The new printed log is here:

> The PNG file(/tmp/test15179029552538249317image) cannot be decoded. This may be due to an OpenJDK issue with certain PNG files. See https://github.com/robolectric/robolectric/issues/6812 for more details.

We can check e.getCause().getMessage() content for "Invalid chunk length". Those two methods can be null, so there are some null-checking code.